cvmfs/catalog_mgr_impl.h

```diff
@@ -285,7 +285,6 @@ bool AbstractCatalogManager<CatalogT>::LookupPath(const PathString &path,
     LogCvmfs(kLogCatalog, kLogDebug, "looking up '%s' in a nested catalog",
              path.c_str());
     StageNestedCatalog(path, best_fit, false /* is_listable */);
-    Unlock();
```

Comment:
AFAIK one cannot upgrade a read lock to a write lock. The unlock before taking a write lock is necessary.